Add filters for subtotalSuffix and saleBadgeSuffix

opr opened the pull request:
<!-- Start by describing the changes made in this Pull Request, and the reason for such changes. -->
### Based against `fix/3735-filters-api`
- In `CartLineItemRow` execute the `subtotalPriceFormat` filter which extensions can use to to format the `ProductPrice` element that will be displayed under the product name. It must return a string containing `<price/>` so it can be used in `createInterpolateElement`.
- Modify `ProductPrice` so that it outputs the price in a way that follows the `format` it was passed.
- In `CartLineItemRow` execute the `saleBadgePriceFormat` filter which extensions can use to modify the _price_ of the `SaleBadge` component. It does not allow for the `Save` word to be changed currently.
- Add `format` prop to the `SaleBadge` component.
- Render the price in `SaleBadge` using `createInterpolateElement`
- Add tests for `ProductPrice` to test passing a format.
- 
<!-- Reference any related issues or PRs here -->
Fixes #3747
Replaces #3763


<!-- Don't forget to update the title with something descriptive. -->
<!-- If your pull request implements a feature flag, make sure you update [this doc](../docs/blocks/features-and-blocks-behind-a-flag.md) -->


### Screenshots
#### Before
![image](https://user-images.githubusercontent.com/5656702/106171210-34d94480-6189-11eb-8eda-3365a9d1da03.png)

#### After
![image](https://user-images.githubusercontent.com/5656702/106171336-52a6a980-6189-11eb-9475-33d70501f1bd.png)


#### Before 
![image](https://user-images.githubusercontent.com/5656702/107386565-e06c8800-6aeb-11eb-8d61-fa3b7b813e8b.png)

#### After
![image](https://user-images.githubusercontent.com/5656702/107386132-781da680-6aeb-11eb-981a-4e1794ac9f11.png)

<!-- If your change has a visual component, add a screenshot here. A "before" screenshot would also be helpful. -->

### How to test the changes in this Pull Request:

1. Check out add/filter-subtotal-string-2 in blocks.
2. Add a non-subscription product to your cart with a sale price.
3. Add a subscription product to your cart with a sale price.
4. Add varying lengths/types of subscription product to your cart, every 2 weeks, every 6 months etc.
5. Open the cart block and verify the sale badge shows the period for subscriptions products, but not for regular products.
6. Verify the subtotal for subscription products shows under each product name and contains the period.
7. Verify the non-subscription product does not show a period for the subtotal.
<!-- If you can, add the appropriate labels -->

opr committed:
Apply filters for subtotal and sale badge in CartLineItemRow

opr committed:
Append suffix to prices and SaleBadge if passed in as a prop

opr committed:
Add subtotal filter to OrderSummaryItem

opr committed:
Add tests for OrderSummaryItem and ProductPrice

opr committed:
Rename test for ProductPrice

opr committed:
Update checkout filter registration plugin name

opr committed:
Remove obsolete snapshop

opr committed:
Remove suffix and change to format

This is because we want to make the entire price string editable, not just the part after it.

opr committed:
Change formatting on SaleBadge to use createInterpolateElement

opr committed:
Remove tests that are no longer needed.

opr committed:
Update ProductPrice tests to reflect changes to format prop

Aljullu commented on assets/js/base/components/cart-checkout/order-summary/order-summary-item.js:
```diff
@@ -50,6 +51,15 @@ const OrderSummaryItem = ( { cartItem } ) => {
+			lineItem: cartItem,
+		},
+		// Only accept strings.
+		validation: ( value ) => typeof value === 'string',
```
Should we check that it includes `<price/>` too?

opr replied:
Good catch, thanks Albert

budzanowski reviewed (approved):
Looks good.

Aljullu reviewed (approved):
LGTM as well. There is a small visual glitch with the Sale badge under the subtotal overflowing the table:

![imatge](https://user-images.githubusercontent.com/3616980/107505071-7829af00-6b9c-11eb-8a26-56ae198fcdc0.png)

opr committed:
Check that subtotalPriceFormat contains <price/> in OrderSummaryItem

opr committed:
Fix cart block styling to stop badges overflowing the container

opr commented:
> LGTM as well. There is a small visual glitch with the Sale badge under the subtotal overflowing the table:
> 
> ![imatge](https://user-images.githubusercontent.com/3616980/107505071-7829af00-6b9c-11eb-8a26-56ae198fcdc0.png)

Thanks, I had fixed this in the last PR but forgot to copy the CSS changes over into this one. Updated now.
![image](https://user-images.githubusercontent.com/5656702/107508858-832f1000-6b99-11eb-8453-fb3b9fb8f279.png)

Aljullu reviewed (approved):
Looks good now, thanks @opr8!
